### DISCUSSION TITLE
Correct URL to Exchange Token

### DISCUSSION THREAD

**Discussion by zt4** [timdorr/tesla-api#262] [Q&A]
I've followed a bunch of examples on here (mostly https://github.com/timdorr/tesla-api/issues/215#issuecomment-770142656) , and everything is working, until I get down to the last one to exchange the token, which gives me the endpoint deprecated error:

```
POST https://owner-api.teslamotors.com/oauth/token
Authorization: Bearer {access_token}
Content-Type: application/json


{
    grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
    client_id: '81527cff06843c8634fdc09e8ac0abefb46ac849f38fe1e431c2ef2106796384',
    client_secret: 'c7257eb71a564034f9419ee651c7d0e5f7aa6bfbd18bafb5c5c033b093bb2fa3'
}
```

And that gives a response of

`{"response":"endpoint_deprecated:_please_update_your_app."}
`

I tried posting the exchange token request to https://auth.tesla.com/oauth2/v3/token and I get a response of 

`{"error":"invalid_grant","error_description":"Internal server error","error_uri":"https://auth.tesla.com/error/reference/..."}
`

Just want to confirm what URL is working for others?

**Comment by timdorr** [comment]
Are you using the `access_token` from the v3/token request? It should be a short-lived (5 minutes) token that goes into the Authorization header. You know you have the right one if it is a JWT and starts with `eyJ`.

**Reply by zt4** [reply]
I got it working against this URL: https://owner-api.teslamotors.com/oauth/token

I missed including the damn JSON payload in the post... Oops!
